src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/EnclaveDelegate.CngCryto.cs

```diff
@@ -95,6 +60,68 @@ internal byte[] GetSerializedAttestationParameters(SqlEnclaveAttestationParamete
+        }
+
+        private void GetEnclaveSession(SqlConnectionAttestationProtocol attestationProtocol, string enclaveType, EnclaveSessionParameters enclaveSessionParameters, bool generateCustomData, out SqlEnclaveSession sqlEnclaveSession, out long counter, out byte[] customData, out int customDataLength, bool throwIfNull)
+        {
+            SqlColumnEncryptionEnclaveProvider sqlColumnEncryptionEnclaveProvider = GetEnclaveProvider(attestationProtocol, enclaveType);
+            sqlColumnEncryptionEnclaveProvider.GetEnclaveSession(enclaveSessionParameters, generateCustomData, out sqlEnclaveSession, out counter, out customData, out customDataLength);
+
+            if (throwIfNull && sqlEnclaveSession == null)
+            {
+                throw SQL.NullEnclaveSessionDuringQueryExecution(enclaveType, enclaveSessionParameters.AttestationUrl);
+            }
+        }
+
+        internal void InvalidateEnclaveSession(SqlConnectionAttestationProtocol attestationProtocol, string enclaveType, EnclaveSessionParameters enclaveSessionParameters, SqlEnclaveSession enclaveSession)
+        {
+            SqlColumnEncryptionEnclaveProvider sqlColumnEncryptionEnclaveProvider = GetEnclaveProvider(attestationProtocol, enclaveType);
+            sqlColumnEncryptionEnclaveProvider.InvalidateEnclaveSession(enclaveSessionParameters, enclaveSession);
+        }
+
+
+        private SqlColumnEncryptionEnclaveProvider GetEnclaveProvider(SqlConnectionAttestationProtocol attestationProtocol, string enclaveType)
+        {
+            if (!s_enclaveProviders.TryGetValue(attestationProtocol, out SqlColumnEncryptionEnclaveProvider sqlColumnEncryptionEnclaveProvider))
+            {
+                switch (attestationProtocol)
+                {
+                    case SqlConnectionAttestationProtocol.AAS:
+                        AzureAttestationEnclaveProvider azureAttestationEnclaveProvider = new AzureAttestationEnclaveProvider();
+                        s_enclaveProviders[attestationProtocol] = azureAttestationEnclaveProvider;
+                        sqlColumnEncryptionEnclaveProvider = s_enclaveProviders[attestationProtocol];
+                        break;
+
+                    case SqlConnectionAttestationProtocol.HGS:
+                        HostGuardianServiceEnclaveProvider hostGuardianServiceEnclaveProvider = new HostGuardianServiceEnclaveProvider();
+                        s_enclaveProviders[attestationProtocol] = hostGuardianServiceEnclaveProvider;
+                        sqlColumnEncryptionEnclaveProvider = s_enclaveProviders[attestationProtocol];
+                        break;
+
+#if ENCLAVE_SIMULATOR
+                    case SqlConnectionAttestationProtocol.SIM:
+                        SimulatorEnclaveProvider simulatorEnclaveProvider = new SimulatorEnclaveProvider();
+                        EnclaveProviders[attestationProtocol] = (SqlColumnEncryptionEnclaveProvider)simulatorEnclaveProvider;
```

Comment:
Please change this to `s_enclaveProviders`.

Reply:
Done.